src/ui/windows/TogglDesktop/TogglDesktop/ui/ViewModels/TimelineViewModel.cs

```diff
@@ -172,21 +171,25 @@ private enum TimeStampType
             End,
             Empty
         }
-        private static List<TimeEntryBlock> ConvertTimeEntriesToBlocks(List<Toggl.TogglTimeEntryView> timeEntries, int selectedScaleMode, DateTime selectedDate)
+        private List<Toggl.TogglTimeEntryView> ConvertTimeEntriesToBlocks(List<Toggl.TogglTimeEntryView> timeEntries, Toggl.TogglTimeEntryView? runningEntry, int selectedScaleMode, DateTime selectedDate)

```

Comment:
I'm not sure I understand your last point. As for everything else:

- It's still possible to leave this method static like it was before, but I thought this would be a complication to anyone who would read this. Here we need to process all the time entries and the running one all together to have the correct horizontal layout. And then there is an option to return a tuple of 2: list of time entry blocks and running TE. But I thought it's clearer to turn method into non-static and assign 2 properties there.
- The return value is one of its input parameters because `GenerateGapTimeEntries` is subscribed to its observable, I just didn't want to change its signature.

Reply:
My last point is: if you comment out line 61 then this method won't execute. This makes little sense - gap generation shouldn't influence the main blocks generation. The subscription itself is only on line 61.

As for these two:
- Making a method `static` makes it easier to read - if it's static then the reader knows that the method doesn't modify any fields or properties, it's just doing some calculations, etc. It's easier to reason about the method as a whole. While if it's not `static` it can mutate any state - then it means it's necessary to read the method fully to understand what kind of state it mutates.
- If `GenerateGapTimeEntries` needs only `timeEntries` list - an input to the `ConvertTimeEntriesToBlocks` method - why does it need to subscribe to the result of `ConvertTimeEntriesToBlocks`?

Reply:
From what I see, `ConvertTimeEntriesToBlocks` doesn't need to return a tuple, you can make it return a `blocks` list. Then in the separate statements you can define `TimeEntryBlocks` and `RunningTimeEntryBlock`.

Reply:
Hm, I now really don't understand how it happens, that `ConvertTimeEntriesToBlocks` depend on `GenerateGapTimeEntryBlocks` (but I checked, it really does). In my head this was vice versa - `ConvertTimeEntriesToBlocks` performs always when an obsercable from `CombineLatest` send a new value, and `GenerateGapTimeEntryBlocks` would always execute after `ConvertTimeEntriesToBlocks`.

Reply:
`ConvertTimeEntriesToBlocks` doesn't get called without `GenerateGapTimeEntryBlocks` because there's no subscription this way. The subscription happens only inside `ToPropertyEx` on line 61. 
`CombineLatest` doesn't "send" values - it just transforms incoming `IObservable` into a different `IObservable`, but nothing really executes without a subscription.